code/modules/mob/living/basic/boss/thing/thing.dm

```diff
@@ -90,6 +106,36 @@
 	animate(alpha = 0, time = 0.5 SECONDS)
 	SEND_SIGNAL(src, COMSIG_MEGAFAUNA_THETHING_PHASEUPDATED)
 
+/// Delete our return timer when we gain a target if we started premapped
+/mob/living/basic/boss/thing/proc/target_gained(datum/source)
+	SIGNAL_HANDLER
+	if(!return_timer)
+		return
+	deltimer(return_timer)
+	return_timer = null
+
+/// If we started premapped, and we lost our target, start a 3 minute timer to return to spawn turf unless we gain aggro again
+/mob/living/basic/boss/thing/proc/target_lost(datum/source)
+	SIGNAL_HANDLER
+	if(stat || client || loc == spawn_loc || return_timer)
+		return
+	return_timer = addtimer(CALLBACK(src, PROC_REF(return_to_spawn_check)), 3 MINUTES, TIMER_STOPPABLE | TIMER_DELETE_ME)
+
+/// Return us to our spawn loc (ruin boss only) if we are alive and have an ai controller and our loc isnt the spawn loc
+/mob/living/basic/boss/thing/proc/return_to_spawn_check()
+	if(isnull(ai_controller) || QDELETED(src) || loc == spawn_loc || stat || client)
+		return
+	return_to_spawnloc()
+
+/mob/living/basic/boss/thing/proc/return_to_spawnloc()
+	if(isnull(spawn_loc))
+		CRASH("The Thing tried to return to spawn_loc but it was null! This shouldnt happen")
+	for(var/turf/open/target in RANGE_TURFS(1, loc))
+		new /obj/effect/temp_visual/mook_dust(target)
+	playsound(loc, 'sound/effects/meteorimpact.ogg', 40, TRUE)
+	visible_message(span_danger("[src] retreats through the ground back to where it came from!"))
+	forceMove(spawn_loc)
```

Comment:
This is a decent solution, but are you sure there's no better way to prevent it from murdering the entire station or getting lured out and killed by miners? Just feels a bit out of the line with other megafauna

Reply:
It only happens after 3 whole minutes gone without a target.